### PR TITLE
Don't use reader in ModelGraph and layers

### PR DESCRIPTION
**Pull request by vloncar**
# Description

Currently, adding weights data happens during initialization of layers by using the supplied reader. This is a relic of old design that required splitting this from converter. Upcoming ONNX support and nested model require this step to be done during conversion. This PR removes reader from `ModelGraph` and its subsequent use to read weights data. This is now moved to converters themselves and weights data is available as an attribute.

## Type of change

- [x] (not exactly) New feature (non-breaking change which adds functionality)

## Tests

This just changes the existing functionality so existing tests should verify that it works as expected

## Checklist

- [x] I have read the [guidelines for contributing](https://github.com/fastmachinelearning/hls4ml/blob/main/CONTRIBUTING.md).
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have installed and run `pre-commit` on the files I edited or added.

### CODE REVIEW

**Comment by jmitrevs** [comment]
Is this ready for review? My initial impression is positive, but I need to look more carefully.

**Comment by jmitrevs** [comment]
`onnx_to_hls`, `pytorch_to_hls`, and `tf_to_hls` still pass the reader to ModelGraph. They will no longer work with this change. Are they not changed because others (e.g. me for ONNX) will fix this and changing it will make a merge conflict?

By the way, what's the status of tf_to_hls? Does anyone support it?

**Comment by vloncar** [comment]
We agreed that the changes to onnx and pytorch will come after this merge, since both need to be updated. tf_to_hls is not supported anymore, no one uses it, it relies on old v1 compatibility so it is not worth maintaining in its current form.

**Comment by jmitrevs** [comment]
Sounds good. I will make a few more checks and likely merge this. As a separate PR we may consider removing tf_to_hls afterwards.

File: hls4ml/converters/keras_to_hls.py [hunk 1]
```diff
@@ -39,15 +44,28 @@ def get_weights_data(self, layer_name, var_name):
...
             return None
 
-    def get_weights_shape(self, layer_name, var_name):
-        data = self._find_data(layer_name, var_name)
-        if data is not None:
-            return data.shape
+
+class KerasNestedFileReader(KerasFileReader):
+    def __init__(self, data_reader, nested_path):
+        self.config = data_reader.config
```
[the remark this answers is not in the document]
Comment (vloncar) [reply hls4ml/converters/keras_to_hls.py:50]: `KerasNestedFileReader` is actually there by accident, it belongs to the followup PR and does nothing in this PR. I agree with you about the style, but can I fix it in the follow-up?

**Review by jmitrevs — APPROVED** [review]

**Pull request merged**